lnwallet/musession/manager.go

```diff
@@ -0,0 +1,308 @@
+package musession
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// MuSig2State is a struct that holds on to the internal signing session state
+// of a MuSig2 session.
+type MuSig2State struct {
+	// MuSig2SessionInfo is the associated meta information of the signing
+	// session.
+	input.MuSig2SessionInfo
+
+	// context is the signing context responsible for keeping track of the
+	// public keys involved in the signing process.
+	context *musig2.Context
+
+	// session is the signing session responsible for keeping track of the
+	// nonces and partial signatures involved in the signing process.
+	session *musig2.Session
+}
+
+// PrivKeyFetcher is used to fetch a private key that matches a given key desc.
+type PrivKeyFetcher func(*keychain.KeyDescriptor) (*btcec.PrivateKey, error)
+
+// Manager houses the state needed to manage concurrent musig sessions. Each
+// session is identified by a unique session ID which is used by callers to
+// interact with a given session.
+type Manager struct {
+	sync.Mutex
+
+	keyFetcher PrivKeyFetcher
+
+	musig2Sessions map[input.MuSig2SessionID]*MuSig2State
+}
+
+// NewManager creates a new musig manager given an abstract key fetcher.
+func NewManager(keyFetcher PrivKeyFetcher) *Manager {
+	return &Manager{
+		keyFetcher:     keyFetcher,
+		musig2Sessions: make(map[input.MuSig2SessionID]*MuSig2State),
+	}
+}
+
+// MuSig2CreateSession creates a new MuSig2 signing session using the local
+// key identified by the key locator. The complete list of all public keys of
+// all signing parties must be provided, including the public key of the local
+// signing key. If nonces of other parties are already known, they can be
+// submitted as well to reduce the number of method calls necessary later on.
+//
+// The set of sessionOpts are _optional_ and allow a caller to modify the
+// generated sessions. As an example the local nonce might already be generated
+// ahead of time.
+func (m *Manager) MuSig2CreateSession(keyLoc keychain.KeyLocator,
```

Comment:
Hmm the refactor here is pretty much the move. It can't be moved in isolation, since all these methods are a part of the wallet struct as is. Maybe check the diff again as things have shifted due to rebases, etc.